### PR TITLE
Misc DB Fixes

### PR DESCRIPTION
**Pull request by NickTyrer**

### CODE REVIEW

File: sql/migrations/20230623143144_world.sql [hunk 1]
```diff
@@ -0,0 +1,1273 @@
...
+-- All Humanoids Should Swim
+UPDATE `creature_template` SET `inhabit_type` = 3 WHERE `type` = 7 AND `inhabit_type` = 1;
+
+-- ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+-- Holy Spring Does Not Require Quest To Loot
+UPDATE `gameobject_loot_template` SET `ChanceOrQuestChance` = 100 WHERE `entry` = 938 AND `item` = 737;
+
+-- ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+-- Remove Incorrect Creatures
+DELETE FROM `creature` WHERE `guid` IN (21707, 21708);
```
Comment (Daribon) [inline sql/migrations/20230623143144_world.sql]: Why was 2/3 Highperch Soarer removed?
There should be two of them in that area, sniffed waypoints even got added in this commit: https://github.com/vmangos/core/blob/f840413d743d41c4a12578ca54a44c444160fb03/sql/migrations/20230726202837_world.sql#L30

21706 should be deleted, but not 21708.

**Comment by ratkosrb** [comment]
@NickTyrer I have reviewed this pull request and merged all the good parts, while leaving the parts I have a problem with. Please read my comments.

**Comment by NickTyrer** [comment]
@ratkosrb Thanks for the merge. I will go through the remaining issues and update where necessary.

**Comment by NickTyrer** [comment]
@ratkosrb I have reviewed your suggestions and the only disagreement I have is that the "donation" quest progression was not changed during vanilla and the corrections in this PR are correct.

**Comment by NickTyrer** [comment]
@ratkosrb If you have no objections, I will merge this.

File: sql/migrations/20230623143144_world.sql [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+UPDATE `creature_template` SET `loot_id` = 0 WHERE `entry` = 235;
+DELETE FROM `creature_loot_template`  WHERE `entry` = 235;
+
+-- ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+-- Remove Incorrect Creature
+DELETE FROM `creature` WHERE `guid` = 21706;
+
+-- ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+-- Ran Bloodtooth's Skull Should Always Drop
+UPDATE `creature_loot_template` SET `ChanceOrQuestChance` = 100 WHERE `item` = 5388;
```
Comment (ratkosrb) [inline sql/migrations/20230623143144_world.sql:152]: Did you test this on classic to confirm it should be like that?

Comment (NickTyrer) [reply sql/migrations/20230623143144_world.sql:152]: See https://youtu.be/MMNpCZbWZ8c?t=547

**Pull request merged**